packages/flutter_tools/lib/src/ios/simulators.dart

```diff
@@ -580,20 +580,35 @@ class IOSSimulator extends Device {
   }
 }
 
-/// Launches the device log reader process on the host.
-Future<Process> launchDeviceLogTool(IOSSimulator device) async {
-  // Versions of iOS prior to iOS 11 log to the simulator syslog file.
-  if (await device.sdkMajorVersion < 11) {
-    return processUtils.start(<String>['tail', '-n', '0', '-F', device.logFilePath]);
+/// Launches the device log reader process on the host and parses the syslog.
+@visibleForTesting
+Future<Process> launchDeviceSystemLogTool(IOSSimulator device) async {
+  return processUtils.start(<String>['tail', '-n', '0', '-F', device.logFilePath]);
+}
+
+/// Launches the device log reader process on the host and parses unified logging.
+@visibleForTesting
+Future<Process> launchDeviceUnifiedLogging (IOSSimulator device, String appName) async {
+  final StringBuffer predicate = StringBuffer('eventType = logEvent AND ');
+
+  // Either from Flutter or Swift (maybe assertion or fatal error) or from the app itself.
```

Comment:
How about something like:

```dart
String orP(List<String> clauses) => '(${clauses.join(" OR ")})';
String andP(List<String> clauses) => '(${clauses.join(" AND ")})';
String notP(String clause) => 'NOT($clause)';

final String predicate = andP(<String>[
  orP(<String>[
    'senderImagePath ENDSWITH "/Flutter"',
    'senderImagePath ENDSWITH "/libswiftCore.dylib"',
    'processImageUUID == senderImageUUID',
  ]),
  if (appName != null) 'processImagePath ENDSWITH "$appName"',
  notP('eventMessage CONTAINS ": could not find icon for representation -> com.apple."'),
  notP(andP(<String>[
    'eventMessage BEGINSWITH "assertion failed: "',
    'eventMessage CONTAINS " libxpc.dylib "',
  ])),
]);
```

or anything with a bit more structure to it, really. How could it be easier to add new clauses in the future?